### ISSUE TITLE
Linear Regression Bolt's model not learning

### ISSUE THREAD

**Issue by Dheeraj-Arremsetty** [PyTorchLightning/lightning-bolts#575]
## 🐛 Bug

Bolts Linear regression model doesn't learn, loss value is always high. Hyperparameters are not really helping.

Here is my code snippet

```
X, y = load_diabetes(return_X_y=True)
loaders = SklearnDataModule(X, y, val_split=0, test_split=0.2)
train_loader = loaders.train_dataloader(batch_size=50)
test_loader = loaders.test_dataloader(batch_size=50)
checkpoint_callback = ModelCheckpoint()
model = LinearRegression(input_dim=10,output_dim=1, learning_rate=0.00001, bias=False)#diabetes
trainer = pl.Trainer(max_epochs=100, gpus=1, callbacks=[checkpoint_callback], progress_bar_refresh_rate=20)
trainer.fit(model, train_loader)

trainer.test(test_dataloaders=test_loader)
```

Output:
`[{'test_loss': 27038.28125, 'test_mse_loss': 27038.28125}]`

![image](https://user-images.githubusercontent.com/8819938/110149270-7326dc80-7da3-11eb-9044-44e810fa72ac.png)

If we observe loss value is very high, tried training model with more number of epochs from 200, 500, 600, 1000 and also 10,000. But, nothing seems to help reducing loss value.

Below is the screenshot **actual(blue)** vs **predicted(red)**
![image](https://user-images.githubusercontent.com/8819938/110148822-ea0fa580-7da2-11eb-9fa2-04e6dd6c504c.png)

Tried the Boston dataset, the code mention in the documentation. I see a similar issue, where the model doesn't seem to be learning.
Can someone please help me or point me if I am doing something wrong. Bolt's model doesn't really seem to be learning even after changing multiple learning rates, more epochs, and with other hyperparameters.

### To Reproduce

Run the above code snippet

### Environment

 - Google collab

**Comment by annikabrundyn** [comment]
hey @Dheeraj-Arremsetty - thanks for reporting! could you share the colab link to reproduce the issue?

**Comment by Dheeraj-Arremsetty** [comment]
@annikabrundyn Here is the link to collab - https://colab.research.google.com/drive/1zPOJHeCeofwaUorQvSQO4hR6KXxhPAwO?usp=sharing

**Comment by Dheeraj-Arremsetty** [comment]
@annikabrundyn Please let me know if you are able to access the colab notebook and also let me know if you need more info from me.

**Comment by tchaton** [comment]
Dear @annikabrundyn @Dheeraj-Arremsetty,

Thanks for reporting the bug. We will look into it promptly.

Best,
T.C
